stubs/PyYAML/yaml/nodes.pyi

```diff
@@ -1,31 +1,37 @@
-from typing import Any
+from typing import Any, ClassVar
+
+from yaml.error import Mark as Mark
 
 class Node:
-    tag: Any
+    tag: str
     value: Any
-    start_mark: Any
-    end_mark: Any
-    def __init__(self, tag, value, start_mark, end_mark) -> None: ...
+    start_mark: Mark | Any
+    end_mark: Mark | Any
+    def __init__(self, tag: str, value: Any, start_mark: Mark | None, end_mark: Mark | None) -> None: ...
```

Comment:
Unless `value` can't be expressed with the current type system, it should be left unannotated as a mark that this needs work. Same for the classes and functions below.

Reply:
@srittau gladly. Is it a good idea to also add a `def __getattr__(self, name: str) -> Any: ...` to mark the incomplete stubs for `mypy`?

Reply:
Yes please. This is our policy per `CONTRIBUTION.md`. It should also have the `# incomplete` marker:

```python
# For incomplete classes:
class Foo:
    def __getattr__(self, __name: str) -> Any: ...  # incomplete

# For incomplete modules (at the end of the module):
def __getattr__(__name: str) -> Any: ...  # incomplete
```

Reply:
In this context, an "incomplete class" means it's missing attributes or methods. Are there really methods missing? If the types of some arguments or attributes are `Any` or missing, it doesn't make the whole class incomplete in this sense.

Reply:
@Akuli I can't test it right now, but AFAIR `mypy` won't resolve `Node().value` if it's missing in the stubs and report an error about undefined attribute. With the incomplete marker, it will use the actual implementation and report `Node().value` as `Any`. Correct me if I'm wrong though!

Reply:
@Akuli however, [adding `__getattr__` fails stubtest](https://github.com/python/typeshed/runs/2819267479), so I have reverted it for now.

Reply:
Ah, that might be a misunderstanding. While we prefer no annotations for argument and return types (`def foo(x): ...` over `def foo(x: Any) -> Any: ...`), in cases where that's not possible we need to use `Any` (e.g. `x: Any` rather than leaving out `x`). We have some discussion in #5613 about introducing a separate marker type for those cases, though, but for now `Any` will do.

In cases where you want to mark a class as incomplete for other reasons, though, you can `# type: ignore` the `__getattr__()`. (I've amended my example above.)

Reply:
@srittau I see - read your comment wrong, it was only about the types in the method signature. Thank you for the clarification, corrected!